config/modules/acme/default.nix

```diff
@@ -1,19 +1,55 @@
-{ config, ... }:
+{ config, lib, pkgs, ... }:
+
+with lib;
+
 {
-  users.groups.acme = { };
+  options = with types; {
+    security.acme = {
+      nameserver = mkOption {
+        type = str;
+        description = ''
+          Nameserver to push RFC2136 updates to.
+        '';
+      };
+      tsig.key = mkOption {
+        type = str;
+        description = ''
+          Name of the secret key as defined in DNS server configuration.
+        '';
+      };
+      tsig.secret = mkOption {
+        type = str;
+        description = ''
+          Secret key payload.
+        '';
+      };
+    };
+  };
+
+  config = {
+    users.groups.acme = { };
 
-  security.acme = {
-    email = "mail@hackint.org";
-    acceptTerms = true;
-    certs = {
-      "${config.networking.fqdn}" = {
-        extraDomainNames = [
-          "hackint.org"
-          "irc.hackint.org"
-        ];
-        dnsProvider = "rfc2136";
-        credentialsFile = /etc/nixos/secrets/acme/environment;
-        group = "acme";
+    security.acme = {
+      email = "mail@hackint.org";
+      acceptTerms = true;
+      certs = {
+        "${config.networking.fqdn}" = {
+          extraDomainNames = [
+            "hackint.org"
+            "irc.hackint.org"
+          ];
```

Comment:
```suggestion
             "irc.hackint.eu"
             "irc.${countryCode].hackint.org"
             "irc.${region}.hackint.org"
          ];
```

Reply:
What I meant is: some more intelligent generation of this list. Like request those per service? Round-Robin only in the ircd role, hackint.org for the website and matrix, etc.

Reply:
Should this module really have anything that mentions `hackint.org`?  IMO that should be specified in some module that is solely responsible for setting configuration variables.